packages/edit-site/src/components/add-new-template/utils.js

```diff
@@ -32,7 +35,7 @@ import { blockMeta, post, archive } from '@wordpress/icons';
 export const mapToIHasNameAndId = ( entities, path ) => {
 	return ( entities || [] ).map( ( entity ) => ( {
 		...entity,
-		name: decodeEntities( get( entity, path ) ),
+		name: decodeEntities( getValueFromObjectPath( entity, path ) ),
```

Comment:
`path` is expected to be a string, and for `entity` we're handling nullish values just in case. 

https://github.com/WordPress/gutenberg/blob/53d605de19b229b481ac1d745382a3bded470750/packages/edit-site/src/components/add-new-template/utils.js#L245